presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestTpchCostBasedPlan.java

```diff
@@ -0,0 +1,102 @@
+
+import com.facebook.presto.Session.SessionBuilder;
+import com.facebook.presto.sql.analyzer.FeaturesConfig.JoinDistributionType;
+import com.facebook.presto.sql.analyzer.FeaturesConfig.JoinReorderingStrategy;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.tpch.ColumnNaming;
+import com.facebook.presto.tpch.TpchConnectorFactory;
+import com.google.common.collect.ImmutableMap;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static com.facebook.presto.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tpch.TpchConnectorFactory.TPCH_COLUMN_NAMING_PROPERTY;
+import static java.lang.String.format;
+
+/**
+ * This class tests cost-based optimization rules related to joins. It contains unmodified TPCH queries.
+ * This class is using TPCH connector configured in way to mock Hive connector with unpartitioned TPCH tables.
+ */
+public class TestTpchCostBasedPlan
+        extends BaseCostBasedPlanTest
+{
+    /*
+     * CAUTION: The expected plans here are not necessarily optimal yet. Their role is to prevent
+     * inadvertent regressions. A conscious improvement to the planner may require changing some
+     * of the expected plans, but any such change should be verified on an actual cluster with
+     * large amount of data.
+     */
+
+    public TestTpchCostBasedPlan()
+    {
+        super(() -> {
+            String catalog = "local";
+            SessionBuilder sessionBuilder = testSessionBuilder()
+                    .setCatalog(catalog)
+                    .setSchema("sf3000.0")
+                    .setSystemProperty("task_concurrency", "1") // these tests don't handle exchanges from local parallel
+                    .setSystemProperty(JOIN_REORDERING_STRATEGY, JoinReorderingStrategy.AUTOMATIC.name())
+                    .setSystemProperty(JOIN_DISTRIBUTION_TYPE, JoinDistributionType.AUTOMATIC.name());
+
+            LocalQueryRunner queryRunner = LocalQueryRunner.queryRunnerWithFakeNodeCountForStats(sessionBuilder.build(), 8);
+            queryRunner.createCatalog(
+                    catalog,
+                    new TpchConnectorFactory(1, false, false),
+                    ImmutableMap.of(TPCH_COLUMN_NAMING_PROPERTY, ColumnNaming.SIMPLIFIED.name()));
+            return queryRunner;
+        });
+    }
+
+    @Override
+    protected Stream<Query> getQueries()
+    {
+        return IntStream.rangeClosed(1, 22)
+                .mapToObj(i -> format("q%02d", i))
+                .map(queryId -> new Query(queryId, tpchQuery(queryId)));
+    }
+
+    private static Path tpchQuery(String queryId)
+    {
+        return Paths.get(format("./presto-benchto-benchmarks/src/main/resources/sql/presto/tpch/%s.sql", queryId));
```

Comment:
I see what you are doing. You are trying to pull the resources from the other module. Don't do this. Either copy the queries here, or include the module. We already had similar issues with Tempto and Benchto. It works fine from Intellij, sometimes it works fine from Maven, but then it suddenly begins to fail, because some testing environment packages everything to JAR first. Please don't do this.

Reply:
Maybe we could copy them to generator libraries and take them from there, what do you think?

Reply:
Using `presto-benchto-benchmarks` causes cyclic dependencies, I extracte queries to separate module `presto-benchto-queries`